src/topp/DecoyDatabase.cpp

```diff
@@ -165,173 +187,215 @@ class TOPPDecoyDatabase :
     f.writeStart(out);
     FASTAFile::FASTAEntry entry;
 
-    // Configure Enzymatic digestion
-    // TODO: allow user-specified regex
-    ProteaseDigestion digestion;
-    String enzyme = getStringOption_("enzyme").trim();
-    if ((input_type == SeqType::protein) && !enzyme.empty())
-    {
-      digestion.setEnzyme(enzyme);
-    }
-    // check if decoy_string is common decoy string (e.g. decoy, rev, ...)
-    String decoy_string_lower = decoy_string;
-    decoy_string_lower.toLower();
-    bool is_common = false;
-    for (const auto& a : DecoyHelper::affixes)
+
+ // Neighbor peptide search   
+    String neighbor_file = getStringOption_("Neighbor_Search:neighbor_in");
+ // Check if the neighbor file option is provided
+    if (! neighbor_file.empty())
     {
-      if ((decoy_string_lower.hasPrefix(a) && decoy_string_position_prefix) || (decoy_string_lower.hasSuffix(a) && !decoy_string_position_prefix))
+      String resolution = getStringOption_("Neighbor_Search:resolution");
+      double ion_tolerance = getDoubleOption_("Neighbor_Search:ion_tolerance");
+      double mass_tolerance = getDoubleOption_("Neighbor_Search:mass_tolerance");
+ // Create a ProteaseDigestion object for peptide digestion
+      ProteaseDigestion digestion;
+      String enzyme = getStringOption_("enzyme").trim();
+      if ((input_type == SeqType::protein) && ! enzyme.empty()) 
```

Comment:
this is duplicated from the `else {` section below.
You can probably move this up and avoid the duplication.